File: src/main/java/org/iuc/mobileteam/iucnewsappbackend/repositories/AnnouncementsRepository.java
package org.iuc.mobileteam.iucnewsappbackend.repositories;

import org.iuc.mobileteam.iucnewsappbackend.models.AnnouncementsEntity;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface AnnouncementsRepository {
    AnnouncementsEntity save(AnnouncementsEntity announcement);
    AnnouncementsEntity findById(String id);
    List<AnnouncementsEntity> findAll();
    long deleteAll();
    long deleteById(String id);
    List<AnnouncementsEntity> saveAll(List<AnnouncementsEntity> announcements);
    AnnouncementsEntity update(AnnouncementsEntity announcement);
}

File: src/main/kotlin/com/iucmobileteam/iucnewsbackend/repository/AnnouncementsRepository.kt
package com.iucmobileteam.iucnewsbackend.repository

import com.iucmobileteam.iucnewsbackend.entity.AnnouncementsEntity
import org.springframework.data.jpa.repository.JpaRepository

interface AnnouncementsRepository: JpaRepository<AnnouncementsEntity, Long>